src/odin_devices/si5324.py

```diff
@@ -0,0 +1,539 @@
+            10,11,19,
+            20,21,22,23,24,25,
+            31,32,33,34,35,36,
+            40,41,42,43,44,45,46,47,48,
+            55,
+            131,132,137,138,139,
+            142,143,
+            136]                        # Register 136 is here by convention (iCAL trigger)
+
+    # Clock IDs
+    CLOCK_NONE  = 0
+    CLOCK_1     = 1
+    CLOCK_2     = 2
+    CLOCK_X     = 3
+
+    # Autoselection Options:
+    AUTOMODE_Manual             = 0b00
+    AUTOMODE_Auto_Non_Revertive = 0b01
+    AUTOMODE_Auto_Revertive     = 0b10
+
+    # Define control fields within I2C registers
+    _FIELD_Free_Run_Mode = _Field(0,6,1)     #FREE_RUN Free Run Mode Enable
+
+    _FIELD_Clock_1_Priority = _Field(1,1,2)  #CK_PRIOR2 Clock with 2nd priority
+    _FIELD_Clock_2_Priority = _Field(1,3,2)  #CK_PRIOR1 Clock with 1st priority
+
+    _FIELD_Clock_Select = _Field(3,7,2)      #CLKSEL_REG Manual clock selection
+
+    _FIELD_Autoselection = _Field(4,7,2)     #AUTOSEL_REG Autoselection mode
+
+    _FIELD_Clock_Active = _Field(128,1,2)    #CKx_ACTV_REG for clocks 1 and 2
+
+    _FIELD_LOS1_INT = _Field(129,1,1)        #LOS1_INT Loss of Signal alarm for CLKIN_1
+    _FIELD_LOS2_INT = _Field(129,2,1)        #LOS2_INT Loss of Signal alarm for CLKIN_2
+    _FIELD_LOSX_INT = _Field(129,0,1)        #LOSX_INT Loss of Signal alarm for XA/XB
+
+    _FIELD_FOSC1_INT = _Field(130,1,1)       #FOSC1_INT Frequency Offset alarm for CLKIN_1
+    _FIELD_FOSC2_INT = _Field(130,2,1)       #FOSC2_INT Frequency Offset alarm for CLKIN_2
+    _FIELD_LOL_INT = _Field(130,0,1)         #LOL_INT Loss of Lock alarm
+
+    _FIELD_ICAL_TRG = _Field(136,6,1)        #ICAL Internal Calibration Trigger
+    _FIELD_RST_TRG = _Field(136,7,1)          #RST_REG Internal Reset Trigger
+
+    # NOTE: FLGs need manual clearing, for live alarm status, use corresponding INT signals...
+    _FIELD_FOSC1_FLG = _Field(132,2,1)       #FOSC1_FLG Frequency Offset Flag for CLKIN_1
+    _FIELD_FOSC2_FLG = _Field(132,3,1)       #FOSC2_FLG Frequency Offset Flag for CLKIN_2
+    _FIELD_LOL_FLG = _Field(132,1,1)         #LOL_FLG Loss of Lock Flag
+    # TODO define remaining relevant registers
+
+    def __init__(self, address=0x68, **kwargs):
+        """
+        Initialise the SI5324 device.
+
+        :param address: The address of the SI5324 is determined by pins A[2:0] as follows: 0b1101[A2][A1][A0].
+        """
+        I2CDevice.__init__(self, address, **kwargs)
+        logger.info("Created new si5324 instance with address 0x{:02X}.".format(address))
+        self.iCAL_required = True           # An iCAL is required at least once before run
+
+
+    """
+    Utility Functions:
+    """
+
+    @staticmethod
+    def pins_to_address(A2,A1,A0):
+        """
+        Return value of address that self.will be used by the device based on the
+        address pin states A[2:0]. Arguments should be supplied as 1/0.
+        """
+        return (0b1101000 | (A2 << 2) | (A1 << 1) | A0)
+
+
+    """
```

Comment:
Are these "public" methods taking an instance of a "private" class (`_Field`) as an argument?